featuretools/tests/primitive_tests/test_rolling_primitive.py

```diff
@@ -0,0 +1,144 @@
+import pandas as pd
+import pytest
+
+from featuretools.primitives import (
+    RollingCount,
+    RollingMax,
+    RollingMean,
+    RollingMin,
+    RollingSTD
+)
+from featuretools.primitives.utils import _roll_series_with_gap
+
+
+def test_rolling_max(rolling_series_pd):
+    window_length = 5
+    gap = 2
+
+    expected_vals = _roll_series_with_gap(rolling_series_pd,
```

Comment:
Wondering just a little bit about the robustness of these tests. If I'm looking at this correctly we are more or less duplicating the primitive function here and confirming that the actual primitive function returns the same value as the version we have in the test. If we did something to change `_roll_series_with_gap` that causes incorrect results, that would also get propagated here and the test would pass, but hopefully be caught in the tests specific to that function.

Not sure if it is better to hard code the expected values or not? Not a blocker for me one way or the other, just something to consider.

Reply:
Yeah, my goal in the tests for this feature would be to catch any wrong or unexpected rolling behavior in the tests specific to `_roll_series_with_gap`. 

My goal with the primitive tests was to confirm that they behave exactly like `_roll_series_with_gap` so we could catch any differing behavior that could get introduced within the primitive classes themselves. Or, in the case of `RollingCount` where we're specifically making changes to the output of `_roll_series_with_gap`, test just that differing behavior.

With regard to whether we should hardcode the values, my hesitation with doing that here would just be in the fact that it can get confusing to update hardcoded values if/when things do change. I've had situations when I've broken hardcoded tests in a repo I wasn't super familiar with, and it was a pretty big struggle to figure out whether they broke because the hardcoded value should now just be updated, or if it was because my changes had unintended side effects. I have less of a problem with hardcoded values in the `_roll_series_with_gap` tests, though, so if I should be doing a better job of that, happy to make changes.

Reply:
Sounds good to me. Know need to make any changes.